src/main/java/uk/gov/justice/hmpps/prison/service/enteringandleaving/BookingIntoPrisonService.kt

```diff
@@ -76,16 +76,16 @@ class BookingIntoPrisonService(
     // during our transaction.  Other processes / threads will hang waiting for this to complete.
 
     // The current offender alias is linked to the booking with sequence 1
-    val offender = offenderBookingRepository.findLatestOffenderBookingByNomsIdForUpdate(prisonerIdentifier)
-      .map { it.offender }
+    val (offender, offenderBooking) = offenderBookingRepository.findLatestOffenderBookingByNomsIdForUpdate(prisonerIdentifier)
+      .map { Pair(it.offender, it) }
       // if there are no bookings then get the root offender instead
-      .orElseGet { rootOffenderForUpdate(prisonerIdentifier).getOrThrow() }
+      .orElseGet { Pair(rootOffenderForUpdate(prisonerIdentifier).getOrThrow(), null) }
 
-    return newBookingWithoutUpdateLock(offender, requestForNewBooking)
+    return newBookingWithoutUpdateLock(offender, offenderBooking, requestForNewBooking)
```

Comment:
passing in both then stops us from having to get the latest booking for the prisoner.  We can't just pass in the booking since the prisoner might not have any yet.